libs/crypto/keys_manager_impl.cpp

```diff
@@ -20,39 +20,85 @@
 #include "cryptography/ed25519_sha3_impl/internal/sha3_hash.hpp"
 #include "logger/logger.hpp"
 
+#include <algorithm>
 #include <fstream>
+#include <random>
 #include <utility>
 
 using iroha::operator|;
 
 namespace iroha {
   /**
-   * Return function which will try to deserialize specified value to specified
-   * field in given keypair
+   * Return a function which will try deserialize the value to
+   * specified field in given keypair
    * @tparam T - keypair field type
    * @tparam V - value type to deserialize
    * @param field - keypair field to be deserialized
    * @param value - value to be deserialized
-   * @return keypair on success, otherwise nullopt
+   * @return function that will return keypair on success, otherwise nullopt
    */
   template <typename T, typename V>
-  auto deserializeKeypairField(T iroha::keypair_t::*field, const V &value) {
+  auto deserializeKeypairField(T keypair_t::*field, const V &value) {
     return [=](auto keypair) mutable {
-      return iroha::hexstringToArray<T::size()>(value)
-          | iroha::assignObjectField(keypair, field);
+      return hexstringToArray<T::size()>(value)
+          | assignObjectField(keypair, field);
+    };
+  }
+
+  /**
+   * Function for the private key encryption via XOR
+   * @param privkey is a private key
+   * @param pass_phrase is a key for encryption
+   * @return encrypted string
+   */
+  std::string encrypt(const privkey_t &privkey,
+                      const std::string &pass_phrase) {
+    std::string ciphertext(privkey.size(), '\0');
+    const auto pass_size = std::max(1ul, pass_phrase.size());
+
+    for (auto i = 0u; i < privkey.size(); i++) {
+      ciphertext[i] = privkey[i] ^ pass_phrase[i % pass_size];
+    }
+    return ciphertext;
+  }
+
+  /**
+   * Return a function which will try to deserialize and then decrypt private
+   * key via XORing with pass phrase
+   * @param s is an encrypted data from file
+   * @param pass_phrase for decryption
+   * @return function that will set keypair::privkey on successful
+   *         deserialization and decryption
+   */
+  auto deserializedEncrypted(const std::string &s,
+                             const std::string &pass_phrase) {
+    constexpr auto size = privkey_t::size();
+    auto decryption = [=](auto binstr) {
+      std::string key(binstr.size(), '\0');
+      const auto pass_size = std::max(1ul, pass_phrase.size());
+
+      for (auto i = 0u; i < binstr.size(); i++) {
+        key[i] = binstr[i] ^ pass_phrase[i % pass_size];
```

Comment:
Please add TODO with task id in Jira for replacing XOR encryption with something harder to brake.

Reply:
[Here](https://soramitsu.atlassian.net/browse/IR-863) it is, don't really sure that it should be added as a todo